--- a/libav.h	2024-10-07 19:39:05.590133580 +0200
+++ b/libav.h	2024-10-07 19:40:15.023215167 +0200
@@ -174,7 +174,11 @@
 #define lav_codec_tag(s) s->codecpar->codec_tag
 #define lav_sample_rate(s) s->codecpar->sample_rate
 #define lav_bit_rate(s) s->codecpar->bit_rate
+#if LIBAVCODEC_VERSION_MAJOR >= 60
+#define lav_channels(s) s->codecpar->ch_layout.nb_channels
+#else
 #define lav_channels(s) s->codecpar->channels
+#endif
 #define lav_width(s) s->codecpar->width
 #define lav_height(s) s->codecpar->height
 #define lav_profile(s) s->codecpar->profile
@@ -186,7 +190,11 @@
 #define lav_codec_tag(s) s->codec->codec_tag
 #define lav_sample_rate(s) s->codec->sample_rate
 #define lav_bit_rate(s) s->codec->bit_rate
+#if LIBAVCODEC_VERSION_MAJOR >= 60
+#define lav_channels(s) s->codec->ch_layouts->nb_channels
+#else
 #define lav_channels(s) s->codec->channels
+#endif
 #define lav_width(s) s->codec->width
 #define lav_height(s) s->codec->height
 #define lav_profile(s) s->codec->profile
